src/amo/reducers/reviews.js

```diff
@@ -244,6 +252,12 @@ export const addReviewToState = ({
       // This will trigger a refresh from the server.
       [review.userId]: undefined,
     },
+    ratingSummary: {
+      ...state.ratingSummary,
+      // When adding a new rating, reset the cache of rating summaries.
+      // This will trigger a refresh from the server.
+      [review.addonId]: undefined,
```

Comment:
I am just trying to understand this ...so here you set to undefined and then you update this below in 'SET_RATING_SUMMARY'? this forces any update to update this immediately?

Reply:
The pattern we strive for looks like this:

1. When the forthcoming `RatingSummary` component is constructing, it checks `reviewState.ratingSummary[addonId]`
2. If the value is `undefined`, it dispatches `fetchRatingSummary()` and renders a loading screen
3. If the value is **not** `undefined`, it renders the data

The line you commented on is the reducer for `setRating()` or `setInternalRating()`. Those are both actions for creating or modifying a rating. Whenever that happens, we reset the `ratingSummary` so that `RatingSummary` knows to re-fetch it from the API. It keeps us in sync with the API.

If you ever see something that isn't following this pattern, it probably needs fixing.

We had to get creative in the past to work around stale API caching but those days should hopefully be over. The API has evolved.

Reply:
cool, thanks for breaking it down : )